stdlib/bigstring.mli

```diff
@@ -0,0 +1,173 @@
+    byte arrays between OCaml code and C libraries.
+
+    Bigstrings support all the OCaml ad-hoc polymorphic operations:
+    - comparisons ([=], [<>], [<=], etc, as well as {!Pervasives.compare});
+    - hashing (module [Hash]);
+    - and structured input-output (the functions from the
+     {!Marshal} module, as well as {!Pervasives.output_value}
+     and {!Pervasives.input_value}).
+*)
+
+
+(** @since 4.08.0 *)
+
+type t = bigstring
+(** The type of big strings. *)
+
+val create : int -> t
+(** [create n] returns a new big string with size [n]. *)
+
+val init : int -> (int -> char) -> t
+(** [init n f] returns a fresh big string of length [n] with
+    element [i] initialized to the result of [f i] (in increasing index order).
+
+    Raises [Invalid_argument] if [n < 0]. *)
+
+val copy : t -> t
+(** Returns a new big string containing the same bytes as the argument. *)
+
+val length : t -> int
+(** Return the length of the given big string. *)
+
+val get : t -> int -> char
+(** [get a x] returns the element of [a] at index [x].  [x] must be greater or
+    equal than [0] and strictly less than [length a]. Otherwise,
+    [Invalid_argument] is raised. *)
+
+val set : t -> int -> char -> unit
+(** [set a x v] stores the value [v] at index [x] in [a].  [x] must be inside
+    the bounds of [a] as described in {!get}; otherwise, [Invalid_argument] is
+    raised. *)
+
+val unsafe_get : t -> int -> char
+(** Like {!get}, but bounds checking is not always performed.
+    Use with caution and only when the program logic guarantees that
+    the access is within bounds. *)
+
+val unsafe_set : t -> int -> char -> unit
+(** Like {!set}, but bounds checking is not always performed.
+    Use with caution and only when the program logic guarantees that
+    the access is within bounds. *)
+
+val sub : t -> int -> int -> t
+(** [sub s start len] returns a new big string sequence of length [len],
+    containing the subsequence of [s] that starts at position [start] and has
+    length [len]. *)
+
+val sub_nocopy : t -> int -> int -> t
+(** Same as {!sub}, except that the resulting big string shares the same storage
+    as the argument. *)
+
+val sub_bytes : t -> int -> int -> bytes
+(** Same as {!sub}, but returns [bytes]. *)
+
+val sub_string : t -> int -> int -> string
+(** Same as {!sub}, but returns a [string]. *)
+
+val blit : t -> int -> t -> int -> int -> unit
+(** [blit src srcoff dst dstoff len] copies [len] bytes from big string [src] at
+    index [srcidx], to big string [dst], starting at index [dstoff]. It works
+    correctly even if [src] and [dst] are the same big string, and the source
+    and destination intervals overlap.
+
+    Raises [Invalid_argument] if [srcoff] and [len] do not designate a valid
+    range of [src], or if [dstoff] and [len] do not desginate a valid range of
+    [dst]. *)
+
+val blit_bytes : bytes -> int -> t -> int -> int -> unit
+(** Same as {!blit}, but read data from a byte sequence. *)
```

Comment:
You should precise than this function (instead `blit`) use internally `memcpy`.

Reply:
Good point, thanks.